src/fondant/component/executor.py

```diff
@@ -491,14 +492,19 @@ def wrap_transform(
         """
 
         def wrapped_transform(dataframe: pd.DataFrame) -> pd.DataFrame:
-            # Call transform method
-            dataframe = transform(dataframe)
-
-            # Drop columns not in specification
+            # Columns of operation specification
             columns = [
                 name for name, field in operation_spec.operation_produces.items()
             ]
 
+            if dataframe.empty:
+                logger.info("Received empty partition, skipping transformation.")
+                return pd.DataFrame(columns=columns)
+
+            # Call transform method
+            dataframe = transform(dataframe)
```

Comment:
Can we simplify this like this?

```suggestion
            if not dataframe.empty:
                dataframe = transform(dataframe)
            else:
                logger.info("Received empty partition, skipping transformation.")
```